### DISCUSSION TITLE
When I'm planning a route by train, it always chooses the wrong POI.

### DISCUSSION THREAD

**Discussion by CornerSyrup** [osmandapp/OsmAnd#19029] [Q&A] [answered]
This is what the system chooses when I plan with the train station.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/65886108/a83a04c5-e418-4c6c-b9d6-448c56093640"  width="300" height="auto"/> </td>
  </tr>
</table>

Then I have to correct it to the right platform.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/65886108/0719ae53-0b9a-4933-b104-5c679ba9551d"  width="300" height="auto"/> </td>
  </tr>
</table>

I know I shouldn't compare it to Google Map, but it's annoying. Those 2 platforms are so close, and the wrong platform will bring me to the terminal in the opposite direction then come back to the starting platform.

Please tell me what I have done wrong, or choose a platform as the starting point is by design (aka user should never use station as the starting point)

**Comment by osmandlover** [comment, marked as answer]
The train routing isn't very good and is more of a fun  gimmick really. Osmand can only route you along ways, and the train routing only routes you along railway ways. That means first it has to snap you to the nearest acceptable way. It does this for the other routing modes too, if you try a car it will first snap you to the nearest road, if you try a walk it will first snap you to the nearest footpath. 

So when you do the train, the first thing is it snaps you to the nearest railway station stop. Because your starting location is nearer to the top track stop, it's going to snap you there. And then it finds the route from there, even if it has to go the wrong direction first. 

I dont think it's by design that you have to pick a platform. The train routing is just a fun feature with room for improvement!
